http/src/main/java/io/micronaut/http/body/ResponseBodyWriter.java

```diff
@@ -40,24 +40,46 @@ public interface ResponseBodyWriter<T> extends MessageBodyWriter<T> {
     /**
      * Writes an object as a {@link ByteBodyHttpResponse}.
      *
-     * @param bufferFactory The buffer factory
-     * @param request       The request
-     * @param httpResponse  The response
-     * @param type          The response body type
-     * @param mediaType     The media type
-     * @param object        The object to write
+     * @param bodyFactory  The buffer factory
+     * @param request      The request
+     * @param httpResponse The response
+     * @param type         The response body type
+     * @param mediaType    The media type
+     * @param object       The object to write
      * @return A {@link ByteBodyHttpResponse} with the response bytes
      * @throws CodecException If an error occurs encoding
      */
     @NonNull
     ByteBodyHttpResponse<?> write(
-        @NonNull ByteBufferFactory<?, ?> bufferFactory,
+        @NonNull ByteBodyFactory bodyFactory,
         @NonNull HttpRequest<?> request,
         @NonNull MutableHttpResponse<T> httpResponse,
         @NonNull Argument<T> type,
         @NonNull MediaType mediaType,
         T object) throws CodecException;
 
+    /**
+     * Write a <i>piece</i> of a larger response, e.g. when writing a Publisher or a part of a
```

Comment:
What exactly is a `piece` of the response? A piece of the body? Should it be `writebodyPart`?  What is the relation to the returned closeable body? How is supposed to close it?

Reply:
A "piece" can be a part (as in multipart), but right now it's only a piece of a streamed response (e.g. when you have a streamed json array). I called it piece instead of part to avoid giving the impression that this is only for multipart, though I plan to use this for multipart as well.

The returned body is Closeable because ownership transfers to the caller. The caller has to either close the body or use it (e.g. concatenating for the json array use case).

Reply:
is `Chunk` a better name? That is what Netty refers to it is it not?

Reply:
"chunk" is already used in the context `transfer-encoding: chunked`, so it is not really appropriate either